examples/blake2f.rs

```diff
@@ -0,0 +1,1377 @@
+                            .map(|j| {
+                                ctx.internal(format!("v_mid_vd_bit_vec[{}][{}]", i, j).as_str())
+                            })
+                            .collect()
+                    })
+                    .collect();
+                let setup_v_mid_vd_bit_vec = v_mid_vd_bit_vec.clone();
+                let v_mid_vc_bit_vec: Vec<Vec<Queriable<F>>> = (0..V_LEN)
+                    .map(|i| {
+                        (0..BITS_NUMBER + 1)
+                            .map(|j| {
+                                ctx.internal(format!("v_mid_vc_bit_vec[{}][{}]", i, j).as_str())
+                            })
+                            .collect()
+                    })
+                    .collect();
+                let setup_v_mid_vc_bit_vec = v_mid_vc_bit_vec.clone();
+                let v_mid_vb_bit_vec: Vec<Vec<Queriable<F>>> = (0..V_LEN)
+                    .map(|i| {
+                        (0..BITS_NUMBER)
+                            .map(|j| {
+                                ctx.internal(format!("v_mid_vb_bit_vec[{}][{}]", i, j).as_str())
+                            })
+                            .collect()
+                    })
+                    .collect();
+                let setup_v_mid_vb_bit_vec = v_mid_vb_bit_vec.clone();
+
+                let v_xor_d_bit_vec: Vec<Vec<Queriable<F>>> = (0..G_ROUND)
+                    .map(|i| {
+                        (0..BITS_NUMBER)
+                            .map(|j| {
+                                ctx.internal(format!("v_xor_d_bit_vec[{}][{}]", i, j).as_str())
+                            })
+                            .collect()
+                    })
+                    .collect();
+                let setup_v_xor_d_bit_vec = v_xor_d_bit_vec.clone();
+
+                let v_xor_b_bit_vec: Vec<Vec<Queriable<F>>> = (0..G_ROUND)
+                    .map(|i| {
+                        (0..BITS_NUMBER)
+                            .map(|j| {
+                                ctx.internal(format!("v_xor_b_bit_vec[{}][{}]", i, j).as_str())
+                            })
+                            .collect()
+                    })
+                    .collect();
+                let setup_v_xor_b_bit_vec = v_xor_b_bit_vec.clone();
+
+                let b_bit_vec: Vec<Queriable<F>> = (0..G_ROUND / 2)
+                    .map(|i| ctx.internal(format!("b_bit_vec[{}]", i).as_str()))
+                    .collect();
+                let setup_b_bit_vec = b_bit_vec.clone();
+                let b_3bits_vec: Vec<Queriable<F>> = (0..G_ROUND / 2)
+                    .map(|i| ctx.internal(format!("b_3bits_vec[{}]", i).as_str()))
+                    .collect();
+                let setup_b_3bits_vec = b_3bits_vec.clone();
+
+                ctx.setup(move |ctx| {
+                    let s = SIGMA_VALUES[r % 10];
+
+                    // Rotation constants | (R1, R2, R3, R4)  | = | (32, 24, 16, 63)
```

Comment:
@10to4 I think this comment here it is confusing, because you are not using the R* constants right away, and anyway the constants are named already. And the values shouldn't change as we are only doing black2b

Reply:
That's right, I'll remove it.